### ISSUE TITLE
Deprecation warnings from trio 0.20

### ISSUE THREAD

**Issue by mweinelt** [agronholm/anyio#423]
Multiple tests are failing on 3.3.4 with trio 0.20 and the following deprecation warning:

*Just noticed our anyio is out of date, will recheck with 3.5.0*

> E   trio.TrioDeprecationWarning: trio.open_process is deprecated since Trio 0.20.0; use trio.lowlevel.open_process instead (https://github.com/python-trio/trio/issues/1104)

Full log:

```

_________________________ test_run_process[trio-shell] _________________________
tests/test_subprocesses.py:33: in test_run_process
    process = await run_process(command, input=b'abc')
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/_core/_subprocesses.py:43: in run_process
    async with await open_process(command, stdin=PIPE if input else DEVNULL, stdout=stdout,
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/_core/_subprocesses.py:90: in open_process
    return await get_asynclib().open_process(command, shell=shell, stdin=stdin, stdout=stdout,
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/_backends/_trio.py:286: in open_process
    process = await trio.open_process(command, stdin=stdin, stdout=stdout, stderr=stderr,
/nix/store/z6vajbb9lsaha074jk2x4d4rajyqng7d-python3.9-trio-0.20.0/lib/python3.9/site-packages/trio/_deprecate.py:120: in __getattr__
    warn_deprecated(thing, info.version, issue=info.issue, instead=instead)
/nix/store/z6vajbb9lsaha074jk2x4d4rajyqng7d-python3.9-trio-0.20.0/lib/python3.9/site-packages/trio/_deprecate.py:51: in warn_deprecated
    warnings.warn(TrioDeprecationWarning(msg), stacklevel=stacklevel)
E   trio.TrioDeprecationWarning: trio.open_process is deprecated since Trio 0.20.0; use trio.lowlevel.open_process instead (https://github.com/python-trio/trio/issues/1104)
_________________________ test_run_process[trio-exec] __________________________
tests/test_subprocesses.py:33: in test_run_process
    process = await run_process(command, input=b'abc')
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/_core/_subprocesses.py:43: in run_process
    async with await open_process(command, stdin=PIPE if input else DEVNULL, stdout=stdout,
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/_core/_subprocesses.py:90: in open_process
    return await get_asynclib().open_process(command, shell=shell, stdin=stdin, stdout=stdout,
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/_backends/_trio.py:286: in open_process
    process = await trio.open_process(command, stdin=stdin, stdout=stdout, stderr=stderr,
/nix/store/z6vajbb9lsaha074jk2x4d4rajyqng7d-python3.9-trio-0.20.0/lib/python3.9/site-packages/trio/_deprecate.py:120: in __getattr__
    warn_deprecated(thing, info.version, issue=info.issue, instead=instead)
/nix/store/z6vajbb9lsaha074jk2x4d4rajyqng7d-python3.9-trio-0.20.0/lib/python3.9/site-packages/trio/_deprecate.py:51: in warn_deprecated
    warnings.warn(TrioDeprecationWarning(msg), stacklevel=stacklevel)
E   trio.TrioDeprecationWarning: trio.open_process is deprecated since Trio 0.20.0; use trio.lowlevel.open_process instead (https://github.com/python-trio/trio/issues/1104)
________________________ test_run_process_checked[trio] ________________________
tests/test_subprocesses.py:40: in test_run_process_checked
    await run_process([sys.executable, '-c',
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/_core/_subprocesses.py:43: in run_process
    async with await open_process(command, stdin=PIPE if input else DEVNULL, stdout=stdout,
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/_core/_subprocesses.py:90: in open_process
    return await get_asynclib().open_process(command, shell=shell, stdin=stdin, stdout=stdout,
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/_backends/_trio.py:286: in open_process
    process = await trio.open_process(command, stdin=stdin, stdout=stdout, stderr=stderr,
/nix/store/z6vajbb9lsaha074jk2x4d4rajyqng7d-python3.9-trio-0.20.0/lib/python3.9/site-packages/trio/_deprecate.py:120: in __getattr__
    warn_deprecated(thing, info.version, issue=info.issue, instead=instead)
/nix/store/z6vajbb9lsaha074jk2x4d4rajyqng7d-python3.9-trio-0.20.0/lib/python3.9/site-packages/trio/_deprecate.py:51: in warn_deprecated
    warnings.warn(TrioDeprecationWarning(msg), stacklevel=stacklevel)
E   trio.TrioDeprecationWarning: trio.open_process is deprecated since Trio 0.20.0; use trio.lowlevel.open_process instead (https://github.com/python-trio/trio/issues/1104)
_____________________________ test_terminate[trio] _____________________________
tests/test_subprocesses.py:63: in test_terminate
    async with await open_process([sys.executable, str(script_path)]) as process:
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/_core/_subprocesses.py:90: in open_process
    return await get_asynclib().open_process(command, shell=shell, stdin=stdin, stdout=stdout,
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/_backends/_trio.py:286: in open_process
    process = await trio.open_process(command, stdin=stdin, stdout=stdout, stderr=stderr,
/nix/store/z6vajbb9lsaha074jk2x4d4rajyqng7d-python3.9-trio-0.20.0/lib/python3.9/site-packages/trio/_deprecate.py:120: in __getattr__
    warn_deprecated(thing, info.version, issue=info.issue, instead=instead)
/nix/store/z6vajbb9lsaha074jk2x4d4rajyqng7d-python3.9-trio-0.20.0/lib/python3.9/site-packages/trio/_deprecate.py:51: in warn_deprecated
    warnings.warn(TrioDeprecationWarning(msg), stacklevel=stacklevel)
E   trio.TrioDeprecationWarning: trio.open_process is deprecated since Trio 0.20.0; use trio.lowlevel.open_process instead (https://github.com/python-trio/trio/issues/1104)
____________________________ test_process_cwd[trio] ____________________________
tests/test_subprocesses.py:77: in test_process_cwd
    result = await run_process(cmd, cwd=tmp_path)
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/_core/_subprocesses.py:43: in run_process
    async with await open_process(command, stdin=PIPE if input else DEVNULL, stdout=stdout,
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/_core/_subprocesses.py:90: in open_process
    return await get_asynclib().open_process(command, shell=shell, stdin=stdin, stdout=stdout,
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/_backends/_trio.py:286: in open_process
    process = await trio.open_process(command, stdin=stdin, stdout=stdout, stderr=stderr,
/nix/store/z6vajbb9lsaha074jk2x4d4rajyqng7d-python3.9-trio-0.20.0/lib/python3.9/site-packages/trio/_deprecate.py:120: in __getattr__
    warn_deprecated(thing, info.version, issue=info.issue, instead=instead)
/nix/store/z6vajbb9lsaha074jk2x4d4rajyqng7d-python3.9-trio-0.20.0/lib/python3.9/site-packages/trio/_deprecate.py:51: in warn_deprecated
    warnings.warn(TrioDeprecationWarning(msg), stacklevel=stacklevel)
E   trio.TrioDeprecationWarning: trio.open_process is deprecated since Trio 0.20.0; use trio.lowlevel.open_process instead (https://github.com/python-trio/trio/issues/1104)
____________________________ test_process_env[trio] ____________________________
tests/test_subprocesses.py:86: in test_process_env
    result = await run_process(cmd, env=env)
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/_core/_subprocesses.py:43: in run_process
    async with await open_process(command, stdin=PIPE if input else DEVNULL, stdout=stdout,
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/_core/_subprocesses.py:90: in open_process
    return await get_asynclib().open_process(command, shell=shell, stdin=stdin, stdout=stdout,
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/_backends/_trio.py:286: in open_process
    process = await trio.open_process(command, stdin=stdin, stdout=stdout, stderr=stderr,
/nix/store/z6vajbb9lsaha074jk2x4d4rajyqng7d-python3.9-trio-0.20.0/lib/python3.9/site-packages/trio/_deprecate.py:120: in __getattr__
    warn_deprecated(thing, info.version, issue=info.issue, instead=instead)
/nix/store/z6vajbb9lsaha074jk2x4d4rajyqng7d-python3.9-trio-0.20.0/lib/python3.9/site-packages/trio/_deprecate.py:51: in warn_deprecated
    warnings.warn(TrioDeprecationWarning(msg), stacklevel=stacklevel)
E   trio.TrioDeprecationWarning: trio.open_process is deprecated since Trio 0.20.0; use trio.lowlevel.open_process instead (https://github.com/python-trio/trio/issues/1104)
_____________________ test_run_sync_in_process_pool[trio] ______________________
tests/test_to_process.py:27: in test_run_sync_in_process_pool
    worker_pid = await to_process.run_sync(os.getpid)
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/to_process.py:123: in run_sync
    process = await open_process(command, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/_core/_subprocesses.py:90: in open_process
    return await get_asynclib().open_process(command, shell=shell, stdin=stdin, stdout=stdout,
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/_backends/_trio.py:286: in open_process
    process = await trio.open_process(command, stdin=stdin, stdout=stdout, stderr=stderr,
/nix/store/z6vajbb9lsaha074jk2x4d4rajyqng7d-python3.9-trio-0.20.0/lib/python3.9/site-packages/trio/_deprecate.py:120: in __getattr__
    warn_deprecated(thing, info.version, issue=info.issue, instead=instead)
/nix/store/z6vajbb9lsaha074jk2x4d4rajyqng7d-python3.9-trio-0.20.0/lib/python3.9/site-packages/trio/_deprecate.py:51: in warn_deprecated
    warnings.warn(TrioDeprecationWarning(msg), stacklevel=stacklevel)
E   trio.TrioDeprecationWarning: trio.open_process is deprecated since Trio 0.20.0; use trio.lowlevel.open_process instead (https://github.com/python-trio/trio/issues/1104)
________________________ test_identical_sys_path[trio] _________________________
tests/test_to_process.py:34: in test_identical_sys_path
    assert await to_process.run_sync(eval, 'sys.path') == sys.path
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/to_process.py:123: in run_sync
    process = await open_process(command, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/_core/_subprocesses.py:90: in open_process
    return await get_asynclib().open_process(command, shell=shell, stdin=stdin, stdout=stdout,
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/_backends/_trio.py:286: in open_process
    process = await trio.open_process(command, stdin=stdin, stdout=stdout, stderr=stderr,
/nix/store/z6vajbb9lsaha074jk2x4d4rajyqng7d-python3.9-trio-0.20.0/lib/python3.9/site-packages/trio/_deprecate.py:120: in __getattr__
    warn_deprecated(thing, info.version, issue=info.issue, instead=instead)
/nix/store/z6vajbb9lsaha074jk2x4d4rajyqng7d-python3.9-trio-0.20.0/lib/python3.9/site-packages/trio/_deprecate.py:51: in warn_deprecated
    warnings.warn(TrioDeprecationWarning(msg), stacklevel=stacklevel)
E   trio.TrioDeprecationWarning: trio.open_process is deprecated since Trio 0.20.0; use trio.lowlevel.open_process instead (https://github.com/python-trio/trio/issues/1104)
______________________________ test_partial[trio] ______________________________
tests/test_to_process.py:39: in test_partial
    assert await to_process.run_sync(partial(sorted, reverse=True), ['a', 'b']) == ['b', 'a']
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/to_process.py:123: in run_sync
    process = await open_process(command, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/_core/_subprocesses.py:90: in open_process
    return await get_asynclib().open_process(command, shell=shell, stdin=stdin, stdout=stdout,
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/_backends/_trio.py:286: in open_process
    process = await trio.open_process(command, stdin=stdin, stdout=stdout, stderr=stderr,
/nix/store/z6vajbb9lsaha074jk2x4d4rajyqng7d-python3.9-trio-0.20.0/lib/python3.9/site-packages/trio/_deprecate.py:120: in __getattr__
    warn_deprecated(thing, info.version, issue=info.issue, instead=instead)
/nix/store/z6vajbb9lsaha074jk2x4d4rajyqng7d-python3.9-trio-0.20.0/lib/python3.9/site-packages/trio/_deprecate.py:51: in warn_deprecated
    warnings.warn(TrioDeprecationWarning(msg), stacklevel=stacklevel)
E   trio.TrioDeprecationWarning: trio.open_process is deprecated since Trio 0.20.0; use trio.lowlevel.open_process instead (https://github.com/python-trio/trio/issues/1104)
_____________________________ test_exception[trio] _____________________________
tests/test_to_process.py:45: in test_exception
    assert await to_process.run_sync(int, 'a')
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/to_process.py:123: in run_sync
    process = await open_process(command, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/_core/_subprocesses.py:90: in open_process
    return await get_asynclib().open_process(command, shell=shell, stdin=stdin, stdout=stdout,
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/_backends/_trio.py:286: in open_process
    process = await trio.open_process(command, stdin=stdin, stdout=stdout, stderr=stderr,
/nix/store/z6vajbb9lsaha074jk2x4d4rajyqng7d-python3.9-trio-0.20.0/lib/python3.9/site-packages/trio/_deprecate.py:120: in __getattr__
    warn_deprecated(thing, info.version, issue=info.issue, instead=instead)
/nix/store/z6vajbb9lsaha074jk2x4d4rajyqng7d-python3.9-trio-0.20.0/lib/python3.9/site-packages/trio/_deprecate.py:51: in warn_deprecated
    warnings.warn(TrioDeprecationWarning(msg), stacklevel=stacklevel)
E   trio.TrioDeprecationWarning: trio.open_process is deprecated since Trio 0.20.0; use trio.lowlevel.open_process instead (https://github.com/python-trio/trio/issues/1104)
_______________________________ test_print[trio] _______________________________
tests/test_to_process.py:50: in test_print
    worker_pid = await to_process.run_sync(os.getpid)
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/to_process.py:123: in run_sync
    process = await open_process(command, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/_core/_subprocesses.py:90: in open_process
    return await get_asynclib().open_process(command, shell=shell, stdin=stdin, stdout=stdout,
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/_backends/_trio.py:286: in open_process
    process = await trio.open_process(command, stdin=stdin, stdout=stdout, stderr=stderr,
/nix/store/z6vajbb9lsaha074jk2x4d4rajyqng7d-python3.9-trio-0.20.0/lib/python3.9/site-packages/trio/_deprecate.py:120: in __getattr__
    warn_deprecated(thing, info.version, issue=info.issue, instead=instead)
/nix/store/z6vajbb9lsaha074jk2x4d4rajyqng7d-python3.9-trio-0.20.0/lib/python3.9/site-packages/trio/_deprecate.py:51: in warn_deprecated
    warnings.warn(TrioDeprecationWarning(msg), stacklevel=stacklevel)
E   trio.TrioDeprecationWarning: trio.open_process is deprecated since Trio 0.20.0; use trio.lowlevel.open_process instead (https://github.com/python-trio/trio/issues/1104)
___________________________ test_cancel_during[trio] ___________________________
tests/test_to_process.py:74: in test_cancel_during
    worker_pid = await to_process.run_sync(os.getpid)
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/to_process.py:123: in run_sync
    process = await open_process(command, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/_core/_subprocesses.py:90: in open_process
    return await get_asynclib().open_process(command, shell=shell, stdin=stdin, stdout=stdout,
/nix/store/vdq5h07m1qbhxngbkzmiw43kmwvvwgrl-python3.9-anyio-3.3.4/lib/python3.9/site-packages/anyio/_backends/_trio.py:286: in open_process
    process = await trio.open_process(command, stdin=stdin, stdout=stdout, stderr=stderr,
/nix/store/z6vajbb9lsaha074jk2x4d4rajyqng7d-python3.9-trio-0.20.0/lib/python3.9/site-packages/trio/_deprecate.py:120: in __getattr__
    warn_deprecated(thing, info.version, issue=info.issue, instead=instead)
/nix/store/z6vajbb9lsaha074jk2x4d4rajyqng7d-python3.9-trio-0.20.0/lib/python3.9/site-packages/trio/_deprecate.py:51: in warn_deprecated
    warnings.warn(TrioDeprecationWarning(msg), stacklevel=stacklevel)
E   trio.TrioDeprecationWarning: trio.open_process is deprecated since Trio 0.20.0; use trio.lowlevel.open_process instead (https://github.com/python-trio/trio/issues/1104)
```

**Comment by agronholm** [comment]
[This](https://github.com/agronholm/anyio/blob/master/src/anyio/_backends/_trio.py#L43-L46)?

**Comment by mweinelt** [comment]
Fixed in 3.5.0, sorry for the noise.
